### ISSUE TITLE
what is the meaning of several parameters?

### ISSUE THREAD

**Issue by wangpichao** [aimagelab/VKD#5]
what is the meaning of p and k in the training command line?:
python ./tools/train_distill.py mars ./logs/base_mars_resnet50 --exp_name distill_mars_resnet50 --p 12 --k 4 --step_milestone 150 --num_epochs 500

And what relationships between these to parameters with the N and M in your paper?

**Comment by angpo** [comment]
Hi, 

The parameters p and k refer to the batch hard strategy we adopt; namely, we make sure that each batches comprises of K samples drawn from P different identities (therefore, the batch size equals P*K). For a detailed description, please refer to the paper introducing this strategy (["In Defense of the Triplet Loss for Person Re-Identification"](https://arxiv.org/abs/1703.07737), Sec. 2, pag 3.). 

On the other hand, the parameters N and M represent the number of viewpoints given to the teacher and the student respectively. Therefore, in the case of the teacher, each of the P*K batch samples is formed by stacking N frames on the second axis.

A.

**Comment by wangpichao** [comment]
> Hi,
> 
> The parameters p and k refer to the batch hard strategy we adopt; namely, we make sure that each batches comprises of K samples drawn from P different identities (therefore, the batch size equals P*K). For a detailed description, please refer to the paper introducing this strategy (["In Defense of the Triplet Loss for Person Re-Identification"](https://arxiv.org/abs/1703.07737), Sec. 2, pag 3.).
> 
> On the other hand, the parameters N and M represent the number of viewpoints given to the teacher and the student respectively. Therefore, in the case of the teacher, each of the P*K batch samples is formed by stacking N frames on the second axis.
> 
> A.

But in your paper, page 8, implementation details: you set p = 8 with k=4, but in your training command line, you set p=12 with k=4. I think they are not consistent.

**Comment by angpo** [comment]
You are right, we should have been clearer in the paper. However, we always set p=8 during the first step and p=12 during the second one. As the training of the student requires less memory than the training of the teacher (the examples shown to the former comprises of fewer frames, 2 _vs_ 8), we decided to exploit the memory saving increasing p from 8 to 12 (thus speeding-up the training time of distillation). 

A.
